### PR TITLE
Update to the ZMuSkim

### PR DESCRIPTION
**Pull request by rvenditti**
Change the ZMuSkim defintion in order to meet the need of Muons subsystems DPGs.
Main changes wrt to previous version are:
-definition of the "loose" leg to be an isolated track (not a muon as in the previous skim definition)
-apply tight muon ID on the "tight" leg and increase the value of the cut on pT
-apply additional selections on the dimuon candidate (on the dz(loose leg, tight leg),  OS)

New selections have been presented (and blessed) in the muon pog meeting:
https://indico.cern.ch/event/666069/contributions/2722477/attachments/1525026/2384260/ZMUskim_CSC_proposal_Sep18.pdf

### CODE REVIEW

**Comment by fabozzi** [comment]
please test workflow 136.730,136.787

**Comment by drkovalskyi** [comment]
This skim is not tested by Muon POG to satisfy the requirements that we agreed on. Please hold integration until we confirm that it's Ok to proceed.

**Comment by drkovalskyi** [comment]
> This skim is not tested by Muon POG to satisfy the requirements that we agreed on. Please hold integration until we confirm that it's Ok to proceed.

I have checked the implementation and found it reasonable. The rate of the new skim is equal to the rate of current skim. Please proceed with integration of the change.

**Comment by fabozzi** [comment]
+1

File: DPGAnalysis/Skims/python/ZMuSkim_cff.py [hunk 1]
```diff
@@ -1,44 +1,166 @@
 import FWCore.ParameterSet.Config as cms
-
+from Configuration.StandardSequences.MagneticField_cff import *
```
Comment (davidlange6) [inline DPGAnalysis/Skims/python/ZMuSkim_cff.py]: hi @rvenditti - the magnetic field is set at a much higher level than within a skim. Please remove this.

File: DPGAnalysis/Skims/python/ZMuSkim_cff.py [hunk 2]
```diff
@@ -1,44 +1,166 @@
...
+                             src = cms.InputTag("allPatTracks"), 
+                             cut = cms.string("(userIsolation('pat::TrackIso')/pt)<0.4"),
+                             filter = cms.bool(True)
                              )
 
+
+
+
+
+MyPatMuons = cms.EDProducer("PATMuonProducer",
```
Comment (davidlange6) [inline DPGAnalysis/Skims/python/ZMuSkim_cff.py]: please pick a more descriptive name for your producer (presumably one connected with the skim name)

File: DPGAnalysis/Skims/python/ZMuSkim_cff.py [hunk 3]
```diff
@@ -1,44 +1,166 @@
 import FWCore.ParameterSet.Config as cms
-
+from Configuration.StandardSequences.MagneticField_cff import *
 ### HLT filter
 import copy
 from HLTrigger.HLTfilters.hltHighLevel_cfi import *
+from  SimGeneral.HepPDTESSource.pythiapdt_cfi import *
```
Comment (davidlange6) [inline DPGAnalysis/Skims/python/ZMuSkim_cff.py]: i'm not sure I see why this and the next several imports are needed? [I  do see why genericParticleProducer_cfi is needed]

**Comment by fabozzi** [comment]
please test workflow 136.730,136.787

**Comment by fabozzi** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by Dr15Jones** [comment]
This appears to have broken step3 of workflow 1000.0 with the error
```
----- Begin Fatal Exception 11-Oct-2017 01:03:43 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=PATMuonProducer label='tightMuonsCandidateForZMuSkim'
Exception Message:
MissingParameter: Parameter 'recomputeBasicSelectors' not found.
----- End Fatal Exception -------------------------------------------------
```

File: DPGAnalysis/Skims/python/ZMuSkim_cff.py [hunk 4]
```diff
@@ -1,44 +1,166 @@
...
+looseIsoMuonsForZMuSkim = cms.EDFilter("PATGenericParticleSelector",  
+                             src = cms.InputTag("allPatTracks"), 
+                             cut = cms.string("(userIsolation('pat::TrackIso')/pt)<0.4"),
+                             filter = cms.bool(True)
                              )
 
+
+
+###create the "tag collection" of muon candidate, embedding the relevant infos  
+tightMuonsCandidateForZMuSkim = cms.EDProducer("PATMuonProducer",
```
Comment (slava77) [inline DPGAnalysis/Skims/python/ZMuSkim_cff.py]: this has to be cloned and modified from a default config

@gpetruc 
you noticed this issue in #20887

**Comment by slava77** [comment]
looks like an example of merging based on obsolete test results.

**Comment by rvenditti** [comment]
I have updated my branch, do you want me  to open a new PR?
https://github.com/rvenditti/cmssw/blob/ZMuSkim-2017update-CSCProposal/DPGAnalysis/Skims/python/ZMuSkim_cff.py

**Comment by davidlange6** [comment]
hi @rvenditti - yes please make a new PR. Thanks.

**Comment by davidlange6** [comment]
i realized that I could do it - #20741.

**Pull request merged**